homeassistant/components/synology_srm/sensor.py

```diff
@@ -0,0 +1,141 @@
+"""Sensor for Synology SRM routers.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.synology_srm/
+"""
+import synology_srm
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_MONITORED_CONDITIONS,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+CONF_TRAFFIC_INTERVAL = "traffic_interval"
+
+DEFAULT_NAME = "synology_srm"
+DEFAULT_USERNAME = "admin"
+DEFAULT_PORT = 8001
+DEFAULT_SSL = True
+DEFAULT_VERIFY_SSL = False
+
+EXTERNALIP_CONDITION = "core.ddns_extip"
+TRAFFIC_CONDITION = "core.ngfw_traffic"
+
+POSSIBLE_MONITORED_CONDITIONS = {
+    "base.encryption",
+    "base.info",
+    EXTERNALIP_CONDITION,
+    "core.ddns_record",
+    "core.system_utilization",
+    "core.network_nsm_device",
+    TRAFFIC_CONDITION,
+    "mesh.network_wanstatus",
+    "mesh.network_wifidevice",
+    "mesh.system_info",
+}
+
+DEFAULT_MONITORED_CONDITIONS = [
+    EXTERNALIP_CONDITION,
+]
+
+POSSIBLE_TRAFFIC_INTERVAL = {"live", "day", "week", "month"}
+
+DEFAULT_TRAFFIC_INTERVAL = [
+    "live",
+]
+
+PLATFORM_SCHEMA = cv.PLATFORM_SCHEMA.extend(
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

Reply:
Don't like that your policy changes ... and so integrations that have been finished (over one month prior to your policy) no longer make the cut (even they would still work with HA).

Think it would have been better to make all new work have to comply after *insert future date here* will need to comply; rather than just "changing policy today so all stuff that has been submitted now has to comply".

... but TBH will prob just not bother; since it works as I need it to.  Will look into this again in a few months as work's forcing me to take some leave.

Reply:
I agree, should move this platform based copmponent to an integration based.

Will avoid double config for one device.